packages/medusa/src/services/product.ts

```diff
@@ -594,7 +595,7 @@ class ProductService extends TransactionBaseService {
           )
       }
 
-      const result = await this.retrieve(product.id, {
+      const result = await this.withTransaction(manager).retrieve(product.id, {
```

Comment:
todo: line 559 is missing. maybe we can do something like `productServiceTxz = this.withTransaction(manager)` and then use it in both places?

Reply:
@srindom I ve seen that you have merged but there is still a missing transaction here